app/views/machines/show.html.erb

```diff
@@ -1 +1,6 @@
 <h1><%= @machine.location %> Vending Machine</h1>
+<h2>Snacks in My Machine</h2>
+<% @machine.snacks.each do |snack| %>
+  <p><%= snack.name %> Price: <%= snack.price %></p>
+<% end %>
+<p>Average Snack Price: <%= @machine.snack_avg_price %></p>
```

Comment:
take a look at using number_to_currency action view helper to format these prices.